core/primer.md

```diff
@@ -674,3 +674,64 @@ in the entity in order to be model compliant.
 
 Likewise, implementations of the server must validate the entire entity
 against the new model, not just a subset of the entity's attributes.
+
+# Why are `ancestor` attributes that point to themselves allowed?
+
+The [`ancestor` attribute](spec.md#ancestor-attribute) is used to build a
+hierarchy of Versions to facilitate compatibility checking when the
+[`compatibility` attribute](spec.md#compatibility-attribute) is set.
+Generally speaking, implementations are not expected to allow users to
+explicitly create multiple roots for a single Version. However, the
+specification does not mandate this as there may be cases where it is useful
+to allow for multiple roots. In some cases it may even be unavoidable, for
+example when the `maxversions` attribute (See `groups.resources.maxversions`
+under [Resource Model](spec.md#registry-model)) is set to a value that forces
+pruning of the Version tree. In such cases, when deleting the oldest version,
+this could result in a new root being created when there are multiple
+decedents of the deleted Version.
+
+# Creating versions with `singleversionroot` and a `maxversions` limit
+
+When the `groups.resources.maxversions` attribute of the [Resource
+Model](spec.md#registry-model) is set to a specified value for a Resource, the
+server will prune versions when attempting to create a new version that would
+exceed this value. However, there may be cases in which the creation of new
```

Comment:
This paragraph might need to be tweaked because even if maxversions=0 the server could still delete Versions.

Perhaps:
There may be situations where the server will need to prune Versions automatically,
for example to enforce the `maxversions` semantics. In these case if `singleversionroot`
is set to `true` and deleting a Version would result in multiple Versions being roots
then the operation that initiated this situation must be rejected.